app/models/vm_or_template.rb

```diff
@@ -1618,6 +1618,39 @@ def self.vms_by_ipaddress(ipaddress)
     end
   end
 
+  # This creates the following SQL conditional:
+  #
+  #   1 = (SELECT 1
+  #        FROM hardwares
+  #        JOIN networks ON networks.hardware_id = hardwares.id
+  #        WHERE hardwares.vm_or_template_id = vms.id
+  #          AND (networks.ipaddress LIKE "%IPADDRESS%"
+  #               OR networks.ipv6address LIKE "%IPADDRESS%")
+  #        LIMIT 1
+  #       )
+  #
+  # This is simply an existance check, so when one record is found matching the
+  # following conditions:
+  #
+  #   - It is a hardware record that is associated with the vm
+  #   - It has an ipaddress or ipv6address that matches the search
+  #
+  # It will return the VM record.
+  def self.miq_expression_includes_any_ipaddresses_arel(ipaddress)
+    vms       = arel_table
+    networks  = Network.arel_table
+    hardwares = Hardware.arel_table
+
+    match_grouping = networks[:ipaddress].matches("%#{ipaddress}%")
+                       .or(networks[:ipv6address].matches("%#{ipaddress}%"))
```

Comment:
This is very specific.  I wonder if it can be made more generic in the future.

Reply:
Note, this is ok.  I'm just unsure what I would do if we were to try to expand this to other operators/models/fields.

Reply:
Honestly, not sure there is a good way to do this.  To me, it seems like this has to be done on a case by case bases, maybe with some abstraction for certain fields.

Reply:
I think the pattern of `1 = (select 1 from relation where CRITERIA)` is pretty generic. Very common for oracle queries. Wonder if we could do that with some pattern

Reply:
Sure, the `1 =` bit is generic, but not the rest of the `WHERE` clause, which is the bulk of things being defined in this method.

Personally, I think I would rather determine a pattern after we see on become more defined after adding more of these methods, instead of trying to figure out one now.  YAGNI and all of that.

Reply:
Yes, it was more something that looked odd... being so specific for this problem.  We can always make it more generic later.